fladle-plugin/src/main/java/com/osacky/flank/gradle/FulladlePlugin.kt

```diff
@@ -138,3 +45,127 @@ class FulladlePlugin : Plugin<Project> {
     }
   }
 }
+
+fun configureLibraryModule(project: Project, flankGradleExtension: FlankGradleExtension) = project.run {
+  pluginManager.withPlugin("com.android.library") {
+    val fulladleModuleExtension = extensions.getByType(FulladleModuleExtension::class.java)
+    if (fulladleModuleExtension.enabled.get()) {
+      val library = extensions.getByType<LibraryExtension>()
+      library.testVariants.configureEach {
+        if (file("$projectDir/src/androidTest").exists()) {
+
+          val debugApk = fulladleModuleExtension.debugApk.let {
+            buildString {
+              if (it.isPresent) { append("    ") }
+              appendProperty(it, name = "app")
+            }
+          }.trimEnd()
+
+          val maxTestShards = fulladleModuleExtension.maxTestShards.let {
+            buildString {
+              if (it.isPresent) { append("    ") }
+              appendProperty(it, name = "max-test-shards")
+            }
+          }.trimEnd()
+
+          val clientDetails = mapPropertyToYaml(fulladleModuleExtension.clientDetails, "client-details")
+
+          val environmentVariables = mapPropertyToYaml(fulladleModuleExtension.environmentVariables, "environment-variables")
+
+          outputs.configureEach {
+            flankGradleExtension.additionalTestApks.add(
+              project.rootProject.provider {
+                listOf("- test: $outputFile", debugApk, maxTestShards, clientDetails, environmentVariables)
+                  .filter { it.isNotEmpty() }
+                  .joinToString("\n")
+                  .trimStart()
+                  .trimEnd()
+              }
+            )
+          }
+        } else {
+          println("Ignoring $name test variant in $path: No tests in $projectDir/src/androidTest")
+        }
+      }
+    }
+  }
+}
+
+fun configureApplicationModule(project: Project, flankGradleExtension: FlankGradleExtension) = project.run {
+  pluginManager.withPlugin("com.android.application") {
+    val fulladleModuleExtension = extensions.getByType(FulladleModuleExtension::class.java)
+    if (!fulladleModuleExtension.enabled.get())
+      return@withPlugin
+    val appExtension = extensions.getByType<AppExtension>()
```

Comment:
we can probably make the logic more similar in the two cases by using a common base class of `AppExtension` and `LibraryExtension`